app/src/main/java/com/example/dung/music_59/ui/genres/GenresActivity.java

```diff
@@ -24,13 +28,31 @@
         implements TracksAdapter.onClickTrackListener, GenresContract.View {
     private static final String BUNDLE_GRENRE = "BUNDLE_GRENRE";
     private static final String EXTRA_BUNDLE = "EXTRA_BUNDLE";
+    public static MusicService sMusicService;
     private ImageView mImageGenre;
     private RecyclerView mRecyclerTrack;
     private TracksAdapter mTracksAdapter;
     private List<Track> mTracks;
     private GenresContract.Presenter mPresenter;
-    private TrackRepository mRepository;
     private Genre mGenre;
+    private Track mTrack;
+    private Intent mPlayIntent;
+    private boolean mMusicBound = false;
+    public ServiceConnection serviceConnection = new ServiceConnection() {
```

Comment:
```suggestion
    private ServiceConnection mServiceConnection;
```
không khởi tạo ở đây. Nên tạo 1 phương thức boundService().